### PR TITLE
Disambiguate 'message'

### PR DESCRIPTION
**Pull request by andrewgordstewart**
The term 'message' appears overloaded in the context of the wallet-app
relationship:

1. MyApp can "message" MyWallet by calling `Wallet.messageWallet`
2. MyApp can create a "message" for MyWallet by calling any wallet
   function, in wallet-functions, which posts a "message" to the iframe
3. MyWallet can send a message to MyApp by putting a WalletEvent in the wallet's messageOutbox
4. MyApp can message YourApp, where that message could be an
   instruction to message YourWallet (see 1)

The fourth case is the most confusing to me, since the instruction would
then post a message to the wallet, so this action involves three
distinct types of messages.

This PR disambiguates two uses of 'message'
- renaming a "message telling the wallet to do something" to be a instruction.
- renaming "message-to-wallet" as a "wallet-instruction"

### CODE REVIEW

File: packages/magmo-wallet-client/src/wallet-events.ts [hunk 1]
```diff
@@ -241,26 +241,48 @@ export const hideWallet = () => ({
...
 
-// MESSAGING
+// WALLET-TO-WALLET COMMUNICATION
 // =========
 /**
- * The type of event when a message to the opponent's wallet is requested.
+ * The type of event when a message relay to the opponent's wallet is requested.
  */
-export const MESSAGE_REQUEST = 'WALLET.MESSAGING.MESSAGE_REQUEST';
+export const MESSAGE_RELAY_REQUESTED = 'WALLET.MESSAGING.MESSAGE_REQUEST';
```
Comment (kerzhner) [inline packages/magmo-wallet-client/src/wallet-events.ts]: @andrewgordstewart Should the constant value reflect the name?

File: packages/magmo-wallet-client/src/wallet-events.ts [hunk 2]
```diff
@@ -241,26 +241,48 @@ export const hideWallet = () => ({
...
-export const messageRequest = (to: string, data: Commitment | string, signature: string) => ({
-  type: MESSAGE_REQUEST as typeof MESSAGE_REQUEST,
+export const messageRelayRequested = (to: string, data: string) => ({
+  type: MESSAGE_RELAY_REQUESTED as typeof MESSAGE_RELAY_REQUESTED,
   to,
   data,
+});
+
+/**
+ * The event thrown when the wallet requests a message be relayed to the opponent's wallet.
```
Comment (kerzhner) [inline packages/magmo-wallet-client/src/wallet-events.ts]: Not sure about the "thrown" language here. Typically, "throw" is used for errors and exceptions.

File: packages/magmo-wallet-client/src/wallet-events.ts [hunk 3]
```diff
@@ -241,26 +241,48 @@ export const hideWallet = () => ({
...
+) => ({
+  type: COMMITMENT_RELAY_REQUESTED as typeof COMMITMENT_RELAY_REQUESTED,
+  to,
+  commitment,
   signature,
 });
 
 /**
- * The event thrown when the wallet requests a message be sent to the opponent's wallet.
+ * The event thrown when the wallet requests a commitment be relayed to the opponent's wallet.
```
Comment (kerzhner) [inline packages/magmo-wallet-client/src/wallet-events.ts]: Same comment about "thrown"

File: packages/rps/src/redux/message-service/saga.ts [hunk 4]
```diff
@@ -32,29 +36,63 @@ export default function* messageSaga() {
...
 
-interface Message {
+interface CommitmentMessage {
   data: Commitment;
   signature: string;
   queue: Queue;
   userName?: string;
 }
 
+interface MessageMessage {
```
Comment (kerzhner) [inline packages/rps/src/redux/message-service/saga.ts]: Not a big fan of this name. "NonProtocolMessage"? "NonCommitmentMessage"? "MessageForWallet"?

File: packages/rps/src/redux/message-service/saga.ts [hunk 5]
```diff
@@ -32,29 +36,63 @@ export default function* messageSaga() {
...
+  userName?: string;
+}
+
+type Message = CommitmentMessage | MessageMessage;
+
 export function* sendWalletMessageSaga() {
-  const sendMessageChannel = createWalletEventChannel([Wallet.MESSAGE_REQUEST]);
+  const sendMessageChannel = createWalletEventChannel([
+    Wallet.COMMITMENT_RELAY_REQUESTED,
+    MESSAGE_RELAY_REQUESTED,
```
Comment (kerzhner) [inline packages/rps/src/redux/message-service/saga.ts]:
```suggestion
    Wallet.MESSAGE_RELAY_REQUESTED,
```

File: packages/rps/src/redux/message-service/saga.ts [hunk 6]
```diff
@@ -32,29 +36,63 @@ export default function* messageSaga() {
...
+            signature: theirSignature,
+            type: 'Commitment',
+          });
+        } else {
+          yield call(reduxSagaFirebase.database.create, `/messages/${to.toLowerCase()}`, message);
+        }
+        break;
+      }
+      case Wallet.MESSAGE_RELAY_REQUESTED: {
+        const { data, to } = messageRequest;
```
Comment (kerzhner) [inline packages/rps/src/redux/message-service/saga.ts]: What about the http communication case?

File: packages/rps/src/redux/message-service/saga.ts (hunk 4 above)
Comment (andrewgordstewart) [reply packages/rps/src/redux/message-service/saga.ts]: Also not a fan of this name, which is a motivation for this review. The problem is that this is a message to YourApp containing a message to YourWallet. `NonCommitmentMessage` seems the best out of your suggestions, and I don't have a better one.

Comment (kerzhner) [reply packages/rps/src/redux/message-service/saga.ts]: Let's go with `NonCommitmentMessage` for now?

**Pull request merged**